### DISCUSSION TITLE
Flac cue - no songs track list avalible ,how to Fix ?

### DISCUSSION THREAD

**Discussion by masterwishx** [kagemomiji/airsonic-advanced#241] [Q&A] [answered]
i have flac + cue files in folders but no tracks avalible in gui 

![image](https://github.com/kagemomiji/airsonic-advanced/assets/28630321/12b959e5-f7e4-4a39-96e9-533e2fb99d54)

![image](https://github.com/kagemomiji/airsonic-advanced/assets/28630321/7afacf95-64fe-4b89-88ff-e51a11303f4c)

![image](https://github.com/kagemomiji/airsonic-advanced/assets/28630321/7c12f1d1-f179-4837-8868-06375ad96726)

**Comment by masterwishx** [comment]
![image](https://github.com/kagemomiji/airsonic-advanced/assets/28630321/0b0d10cd-bc3c-4559-a57b-5097d8fa709d)

**Reply by kagemomiji** [reply]
@masterwishx 
Could you upload the cue sheet here?  
It's difficult to conduct a detailed investigation as only a part of the cuesheet is visible in the image, and it's unclear what is causing the error.

**Reply by masterwishx** [reply]
REM GENRE Rock
REM DATE 1989
REM DISCID 710A280A
PERFORMER "Кино"
TITLE "Последний герой ( France pour Volya-Production, 1989)"
FILE "Кино - Последний герой  France pour Volya-Production (1989).flac" WAVE
TRACK 01 AUDIO
TITLE "Перемен"
PERFORMER "Кино"
INDEX 00 00:00:00
INDEX 01 00:00:33
TRACK 02 AUDIO
TITLE "Электричка"
PERFORMER "Кино"
INDEX 00 04:55:17
INDEX 01 04:59:00
TRACK 03 AUDIO
TITLE "Война"
PERFORMER "Кино"
INDEX 00 09:47:03
INDEX 01 09:49:35
TRACK 04 AUDIO
TITLE "Троллейбус"
PERFORMER "Кино"
INDEX 00 13:52:02
INDEX 01 13:54:23
TRACK 05 AUDIO
TITLE "Печаль"
PERFORMER "Кино"
INDEX 00 16:48:55
INDEX 01 16:51:63
TRACK 06 AUDIO
TITLE "Последний герой"
PERFORMER "Кино"
INDEX 00 22:00:52
INDEX 01 22:03:18
TRACK 07 AUDIO
TITLE "Группа крови"
PERFORMER "Кино"
INDEX 00 25:08:60
INDEX 01 25:09:63
TRACK 08 AUDIO
TITLE "Мама, мы все сошли с ума"
PERFORMER "Кино"
INDEX 00 29:08:12
INDEX 01 29:12:03
TRACK 09 AUDIO
TITLE "В наших глазах"
PERFORMER "Кино"
INDEX 00 33:00:47
INDEX 01 33:04:00
TRACK 10 AUDIO
TITLE "Спокойная ночь"
PERFORMER "Кино"
INDEX 00 36:49:42
INDEX 01 36:53:63

**Reply by masterwishx** [reply]
another example + comment within :

REM GENRE Rock
REM DATE 1988
REM DISCID 980B240B
REM COMMENT "ExactAudioCopy v1.0b3"
PERFORMER "Кино"
TITLE "Группа Крови (Japan VDP-1511)"
FILE "Кино - Группа Крови (Japan VDP-1511).flac" WAVE
  TRACK 01 AUDIO
    TITLE "Blood Type"
    PERFORMER "Кино"
    INDEX 00 00:00:00
    INDEX 01 00:00:37
  TRACK 02 AUDIO
    TITLE "Shut the Door Behind Me"
    PERFORMER "Кино"
    INDEX 00 04:46:47
    INDEX 01 04:47:47
  TRACK 03 AUDIO
    TITLE "War"
    PERFORMER "Кино"
    INDEX 00 09:02:73
    INDEX 01 09:03:72
  TRACK 04 AUDIO
    TITLE "Good Night"
    PERFORMER "Кино"
    INDEX 00 13:07:50
    INDEX 01 13:08:50
  TRACK 05 AUDIO
    TITLE "We Are All Sick, Mama"
    PERFORMER "Кино"
    INDEX 00 19:15:50
    INDEX 01 19:16:50
  TRACK 06 AUDIO
    TITLE "Boshetunmay"
    PERFORMER "Кино"
    INDEX 00 23:23:04
    INDEX 01 23:24:42
  TRACK 07 AUDIO
    TITLE "In Our Eyes"
    PERFORMER "Кино"
    INDEX 00 27:33:01
    INDEX 01 27:34:00
  TRACK 08 AUDIO
    TITLE "Try to Sing Along"
    PERFORMER "Кино"
    INDEX 00 31:07:55
    INDEX 01 31:08:55
  TRACK 09 AUDIO
    TITLE "Hey You, You Better Keep Walking!"
    PERFORMER "Кино"
    INDEX 00 35:43:63
    INDEX 01 35:44:62
  TRACK 10 AUDIO
    TITLE "It's Our Time, Our Turn!"
    PERFORMER "Кино"
    INDEX 00 39:23:32
    INDEX 01 39:24:70
  TRACK 11 AUDIO
    TITLE "The Legend"
    PERFORMER "Кино"
    INDEX 00 43:20:56
    INDEX 01 43:21:55

**Comment by masterwishx** [comment]
Founded having error getting CUE :

![image](https://github.com/kagemomiji/airsonic-advanced/assets/28630321/c420ae1d-5a34-4e0c-aca1-032970713417)

**Reply by mmskv** [reply]
What airsonic version is that? I get this "Error getting cuesheet" error only once in a 700GB library
![image](https://github.com/kagemomiji/airsonic-advanced/assets/47359245/faa1b532-2f25-4df4-b9af-ede9e8642eda)

**Reply by masterwishx** [reply]
Using latest version, have only 105GB

**Reply by mmskv** [reply]
Go with the stable release then, I have a lot of issues with upstream as well

**Reply by masterwishx** [reply]
checked https://ghcr.io/kagemomiji/airsonic-advanced:stable-latest have same no tracks after [Scan media folders now] from CUE

**Reply by kagemomiji** [reply]
@masterwishx 
I have performed an import using the cuesheet you provided, using test music files, and no errors were encountered. However, to investigate the potential cause of the failure, could you please verify if the "ffprobe" command can successfully retrieve the duration of the files associated with the cuesheet? Additionally, if the duration can be obtained, could you please confirm if it is longer than the start time of the last track specified in the cuesheet?

**Reply by masterwishx** [reply]
Sure I will check it, my database from airsonic advanced old one, maybe I should clear db and scan files again?

**Reply by masterwishx** [reply]
Can you please give example of ffprobe command that I should run?
I don't have much experience in ffmpeg

**Reply by masterwishx** [reply]
also i have my All music files in (/unraid_games... ) - mounted folders in root not in /var/music , is it OK? 
  
![image](https://github.com/kagemomiji/airsonic-advanced/assets/28630321/bd34924a-003d-4484-aae2-32eda28530ca)

**Reply by masterwishx** [reply]
i tryed but have such error:

![image](https://github.com/kagemomiji/airsonic-advanced/assets/28630321/50d50384-3eb7-4ffd-84e7-d0cf0a88e1a3)

**Reply by masterwishx** [reply]
if this is what you needed? 

![image](https://github.com/kagemomiji/airsonic-advanced/assets/28630321/cbbe3805-95a6-454c-b944-3a14e98a69a6)

**Reply by masterwishx** [reply]
@kagemomiji  Yes its 43.21min and longer than start of the last song

**Reply by mmskv** [reply, marked as answer]
I managed to replicate the issue.
```
2023-06-29 17:00:16.478  WARN --- o.a.p.service.MediaFileService           : Error getting cuesheet for /var/music/rock/Кино (Виктор Цой) - Неремастированная CD-дискография (1989-1995)/1989-Последний герой (1989 France pour Volya-Production)/Kino - Le dernier des heroes.cue
```



It was caused by a bad encoding of a cuesheet. Output of `file *.cue` was ```Kino - Le dernier des heroes.cue: ISO-8859 text, with CRLF line terminators```. Thus file was parsed incorrectly

![image](https://github.com/kagemomiji/airsonic-advanced/assets/47359245/30b8aaa6-dc3b-4d62-8e35-cb15f59c02a9)

# Solution
I managed to fix it by rewriting the file as ASCII text (```Kino - Le dernier des heroes.cue: ASCII text```). After that and a rescan the cuesheet was parsed fine

![image](https://github.com/kagemomiji/airsonic-advanced/assets/47359245/9acefcd9-2b2e-4209-85c9-93619af1eefc)

To me this looks more like a bad rip than an airsonic issue.

**Reply by masterwishx** [reply]
I found you are right but in my case its 1251 + win 

What program i can use for reformat cue files to ascii or utf8+unix all at once?

![image](https://github.com/kagemomiji/airsonic-advanced/assets/28630321/2b2ef0f8-1575-410f-a189-6947f97b56b7)

**Reply by masterwishx** [reply]
i changed to utf8 +unix it shows now tracks after rescan but only first track played .

![image](https://github.com/kagemomiji/airsonic-advanced/assets/28630321/f87155f3-6685-4f4b-8687-478641ec23c3)


![image](https://github.com/kagemomiji/airsonic-advanced/assets/28630321/5319c328-c474-447d-8ed6-c6674c02fca0)

**Reply by masterwishx** [reply]
But also %S needs to be added in to (mp3 audio + downgrade sample ) fields as @kagemomiji told me.

**Comment by masterwishx** [comment]
This is not only one that have error ,all of my cue has errors...

**Comment by masterwishx** [comment]
if i need to enable debug or something for check where error please let me know ...

**Comment by masterwishx** [comment]
@kagemomiji 

Thanks to user @mmskv the problem was in wrong file encoding 1251 +win , i changed to utf8 + unix(LF) so now imported after scan but have error on playing only first song on every track ...

![image](https://github.com/kagemomiji/airsonic-advanced/assets/28630321/660212d1-6f46-43b9-8feb-fbf315ba704c)

**Reply by kagemomiji** [reply]
@masterwishx 
From your log , I cannot see any SPLIT options in your transcode command. Could you see https://github.com/kagemomiji/airsonic-advanced/blob/main/docs/troubleshooting.md#troubleshooting and check your transocoding configuration.

**Reply by masterwishx** [reply]
i added %S but now i see nothing in this folder not related to adding %S ...

![image](https://github.com/kagemomiji/airsonic-advanced/assets/28630321/7c019b60-81f8-4ed1-a074-f1230793a175)

![image](https://github.com/kagemomiji/airsonic-advanced/assets/28630321/21eacd5c-5ac6-422b-bd36-f48cef39c00c)

**Reply by masterwishx** [reply]
![image](https://github.com/kagemomiji/airsonic-advanced/assets/28630321/a46bb9cd-f1d9-4a55-a653-48587396ff4a)

**Reply by masterwishx** [reply]
still have first trac playing when press on every track: 

![image](https://github.com/kagemomiji/airsonic-advanced/assets/28630321/99ce409f-025d-4457-946f-6e855a6e7af5)
![image](https://github.com/kagemomiji/airsonic-advanced/assets/28630321/2a7cf83e-eec7-47f4-a62d-4fc0b369362e)

**Reply by kagemomiji** [reply]
@masterwishx Could you also replace `%s` to `%S` at `mp3 audio`?

**Reply by masterwishx** [reply]
Now i have all tracks in flac  0:00 and casnt play at all ,also other flacs that not fixed cant be played

![image](https://github.com/kagemomiji/airsonic-advanced/assets/28630321/7e024ec4-de84-4f6f-888f-c599e672b58d)

![image](https://github.com/kagemomiji/airsonic-advanced/assets/28630321/f98f68ec-44da-44b6-9a89-324be8ebb8ec)

**Reply by masterwishx** [reply]
![image](https://github.com/kagemomiji/airsonic-advanced/assets/28630321/c269d530-46ab-4e0b-bd0f-41aefa573f8c)

**Reply by kagemomiji** [reply]
@masterwishx 
I'm sorry. I told you wrong configuration. Please use `ffmpeg %S -i %s -map 0:0 -b:a %bk -v 0 -f mp3 -` at mp3 audio. 
`%s` parameter is required.

**Reply by masterwishx** [reply]
Thanks its working Now !

**Reply by masterwishx** [reply]
i think i can close it now , can i select @mmskv as answer for wrong encoding in my case ? 
But also %S needs to be added in to (mp3 audio + downgrade sample ) fields  as you told me ...

**Reply by kagemomiji** [reply]
@masterwishx 
I'm fine with it. You can mark mmskv's response as the Best Answer.

**Reply by masterwishx** [reply]
Thanks a lot for your time and help, also for you work to continuing the Project

**Comment by masterwishx** [comment]
I checked the fixed utf8 cue on MPC in Win ,  tracks selection working OK
